shakenfist/db.py

```diff
@@ -235,10 +235,18 @@ def create_snapshot(snapshot_uuid, device, instance_uuid, created):
             'uuid': snapshot_uuid,
             'device': device,
             'instance_uuid': instance_uuid,
-            'created': created
+            'created': created,
+            'status': 'creating'
         })
```

Comment:
I thought this was your error until I expanded the view and realised its mine. Its confusing to me that the path for a snapshot is the time it was created, not the snapshot UUID. That means for every request for a specific snapshot we need to iterate through all the snapshots for that instance. I suspect that's a more common operation than wanting to list them by time? Thoughts?

Reply:
Yes, it does seem a bit strange, but right now listing all the snapshots for an instance is the only way to fetch snapshots. I feel like this should wait until there's a more fleshed out story around snapshots. It also blocks making them an superclass of `baseobject.DatabaseBackedObject` instead of just a dict, since `from_db` can't be reasonably implemented in a sane way.

Reply:
Yeah, I feel like most (all?) things that land in the database should eventually become children of DatabaseBackedObject, but it is fair to say we're going to need to take some time to get there.